core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -499,13 +548,14 @@ public TableMetadata updatePartitionSpec(PartitionSpec newPartitionSpec) {
     }
 
     return new TableMetadata(null, formatVersion, uuid, location,
-        lastSequenceNumber, System.currentTimeMillis(), lastColumnId, schema, newDefaultSpecId,
+        lastSequenceNumber, System.currentTimeMillis(), lastColumnId, currentSchemaId, schemas, newDefaultSpecId,
         builder.build(), Math.max(lastAssignedPartitionId, newPartitionSpec.lastAssignedFieldId()),
-        defaultSortOrderId, sortOrders, properties,
-        currentSnapshotId, snapshots, snapshotLog, addPreviousFile(file, lastUpdatedMillis));
+        defaultSortOrderId, sortOrders, properties, currentSnapshotId, snapshots, snapshotLog,
+        addPreviousFile(file, lastUpdatedMillis));
```

Comment:
It looks like this change wasn't needed and just moves 3 variables?

Reply:
Sorry about these, I got confused on which changes were mine when resolving merge conflicts, I'll be more careful next time.